roles/cloudman-boot/tasks/auth.yaml

```diff
@@ -0,0 +1,211 @@
+- name: Create namespace for keycloak
+  command: /usr/local/bin/kubectl create namespace keycloak
+  ignore_errors: True
+
+- name: Set Keycloak Operator path
+  set_fact: kc_path="/tmp/keycloak-operator-{{ cm_keycloak_operator_version }}"
+
+- name: Check if keycloak operator is already unarchived
+  stat:
+    path: "{{ kc_path }}"
+  register: kc_dir
+
+- name: Download and operator
+  when: not (kc_dir.stat.exists and kc_dir.stat.isdir)
+  block:
+    - name: Download and unarchive keycloak-operator release
+      ansible.builtin.unarchive:
+        src: https://codeload.github.com/{{ cm_keycloak_operator_fork }}/tar.gz/{{ cm_keycloak_operator_version }}
+        dest: /tmp/
+        remote_src: yes
+      when: not (cm_keycloak_operator_version in ["latest", "master"])
+
+    - name: Clone master from github
+      ansible.builtin.git:
+        repo: "https://github.com/{{ cm_keycloak_operator_fork }}"
+        dest: "{{ kc_path }}"
+      when: cm_keycloak_operator_version in ["latest", "master"]
+
+    - name: Go to keycloak operator and prepare the cluster resources
+      command: chdir={{ kc_path }} make cluster/prepare
+
+    - name: Deploy the operator
+      command: /usr/local/bin/kubectl apply -f {{ kc_path }}/deploy/operator.yaml -n {{ cm_keycloak_namespace }}
+
+- name: Render Keycloak instance CRD
+  template:
+    src: keycloakcrd.yml.j2
+    dest: "{{ kc_path }}/deploy/cloudman-keycloak.yaml"
+
+- name: Apply the rendered Keycloak instance
+  command: /usr/local/bin/kubectl apply -f {{ kc_path }}/deploy/cloudman-keycloak.yaml -n {{ cm_keycloak_namespace }}
+
+- name: Render GVL Keycloak realm CRD
+  template:
+    src: keycloak_gvl_realm.yml.j2
+    dest: "{{ kc_path }}/deploy/gvl-keycloakrealm.yaml"
+
+- name: Apply the rendered GVL realm
+  command: /usr/local/bin/kubectl apply -f {{ kc_path }}/deploy/gvl-keycloakrealm.yaml -n {{ cm_keycloak_namespace }}
+
+- name: Render Keycloak ingress cert patch
+  template:
+    src: keycloak_ingress_patch.yml.j2
+    dest: "{{ kc_path }}/deploy/cloudman-keycloak-ingress-patch.yaml"
+  tags: auth
+
+- name: Check if Rancher SAML Client Certificate is already created
```

Comment:
I think it's worth moving the rancher-auth-integration stuff to a separate file?

Reply:
I was originally doing that, but it does make it overall slower with not being able to alternate. Right now i'm doing things like generating the rancher SAML client while waiting for the operator to reconcile the instance and create the deployment/ingress. It adds a few minutes if we separate them and wait for keycloak first then install rancher and wait for it which is what I was originally doing. I thought the block nesting would be enough for the visual understanding. I was also thinking maybe it'd make sense to start tagging things more aggressively to make it easier to run only a specific logical grouping of tasks when we want to? But the latter should probably be a separate PR adding tags throughout the playbook

Reply:
Ok sounds good. Only some of it is block nested though?